x/zoneconcierge/keeper/ibc_header_decorator.go

```diff
@@ -0,0 +1,105 @@
+package keeper
+
+import (
+	"github.com/cometbft/cometbft/crypto/tmhash"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types" //nolint:staticcheck
+	ibctmtypes "github.com/cosmos/ibc-go/v8/modules/light-clients/07-tendermint"
+
+	"github.com/babylonlabs-io/babylon/x/zoneconcierge/types"
+)
+
+var _ sdk.PostDecorator = &IBCHeaderDecorator{}
+
+type IBCHeaderDecorator struct {
+	k Keeper
+}
+
+// NewIBCHeaderDecorator creates a new IBCHeaderDecorator
+func NewIBCHeaderDecorator(k Keeper) *IBCHeaderDecorator {
+	return &IBCHeaderDecorator{
+		k: k,
+	}
+}
+
+func (d *IBCHeaderDecorator) getHeaderAndClientState(ctx sdk.Context, m sdk.Msg) (*types.HeaderInfo, *ibctmtypes.ClientState) {
```

Comment:
can we add some documentation on this function on what it does? or maybe only to `func (d *IBCHeaderDecorator) PostHandle`

Reply:
added docstring for both